exec/csnex/Cell.cs

```diff
@@ -309,6 +313,59 @@ public Boolean Equals(Cell x)
             }
             return false;
         }
+
+        public static void CopyCell(Cell dest, Cell source)
+        {
+            Debug.Assert(source != null);
+            Debug.Assert(dest != null);
+
+            switch (source.type) {
+                case Type.Address:
+                    dest.m_Address = source.Address;
+                    break;
+                case Type.Array:
+                    dest.Array = new List<Cell>(source.Array.Count);
+                    for (int i = 0; i < source.Array.Count; i++) {
+                        dest.Array.Add(FromCell(source.Array[i]));
+                    }
+                    break;
+                case Type.Boolean:
+                    dest.m_Boolean = source.Boolean;
+                    break;
+                case Type.Bytes:
+                    dest.Bytes = new byte[source.Bytes.Length];
+                    System.Array.Copy(source.Bytes, 0, dest.Bytes, 0, source.Bytes.Length);
+                    break;
+                case Type.Dictionary:
+                    dest.m_Dictionary = new SortedDictionary<string, Cell>();
+                    List<string> keys = source.Dictionary.Keys.ToList();
+                    foreach (string k in keys) {
+                        dest.Dictionary.Add(k, FromCell(source.Dictionary[k]));
+                    }
+                    break;
+                case Type.Number:
+                    dest.Number = new Number(source.Number);
+                    break;
+                case Type.Object:
+                    dest.m_Object = source.Object;
+                    break;
+                case Type.String:
+                    dest.String = String.Copy(source.String);
```

Comment:
You don't need to make a copy here because C# `String` [is immutable](https://stackoverflow.com/questions/93091/why-cant-strings-be-mutable-in-java-and-net).

Reply:
Fixed.